onnxruntime/core/providers/cuda/cuda_execution_provider_info.cc

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cuda_execution_provider_info.h"
+
+#include "core/common/string_utils.h"
+#include "core/framework/provider_options_utils.h"
+#include "core/providers/cuda/cuda_common.h"
+
+namespace onnxruntime {
+namespace provider_option_names {
+constexpr const char* kDeviceId = "device_id";
+constexpr const char* kMemLimit = "cuda_mem_limit";
+constexpr const char* kArenaExtendStrategy = "arena_extend_strategy";
+constexpr const char* kCudnnConvAlgo = "cudnn_conv_algo";
+constexpr const char* kDoCopyInDefaultStream = "do_copy_in_default_stream";
+}  // namespace provider_option_names
+
+CUDAExecutionProviderInfo CUDAExecutionProviderInfo::FromProviderOptions(const ProviderOptions& options) {
+  CUDAExecutionProviderInfo info{};
+
+  if (ReadProviderOption(options, provider_option_names::kDeviceId, info.device_id)) {
+    int num_devices = 0;
+    CUDA_CALL_THROW(cudaGetDeviceCount(&num_devices));
+    ORT_ENFORCE(
+        0 <= info.device_id && info.device_id < num_devices,
+        "Invalid ", provider_option_names::kDeviceId, " value: ", info.device_id,
+        ", must be between 0 (inclusive) and ", num_devices, " (exclusive).");
+  }
+  ReadProviderOption(options, provider_option_names::kMemLimit, info.cuda_mem_limit);
```

Comment:
I think it just sets the initial arena size. Maybe it needs a clearer name. @codemzs

Reply:
i think it is the maximum, looks like the value eventually goes to [https://github.com/microsoft/onnxruntime/blob/7a43fa00285ce8a923bf1a1d7907818cdaa0d063/include/onnxruntime/core/framework/allocator.h#L15](https://github.com/microsoft/onnxruntime/blob/7a43fa00285ce8a923bf1a1d7907818cdaa0d063/include/onnxruntime/core/framework/allocator.h#L15)

---
In reply to: [538900994](https://github.com/microsoft/onnxruntime/pull/5923#discussion_r538900994) [](ancestors = 538900994)

Reply:
OK, makes sense then. It looks like an unset value defaults to the numeric limit.